readme.md

```diff
@@ -58,7 +60,51 @@ The second (optional) argument may include the properties below.
  * `createCwd` - `boolean` Create a directory for `process.cwd()`.  This is `true` by default.
  * `createTmp` - `boolean` Create a directory for `os.tmpdir()`.  This is `true` by default.
 
-### Creating files
+### Automatically Creating Files & Directories
```

Comment:
I suggest to clarify this API is to load real local dirs inside mock-fs's fake file system.
I have two doubts:
1. The API name is debatable.
2. I did not see the use cases of mixing mock such as:
```js
mockfs({ real_dir: ..., fake_dir: ... });

mockfs({fake_dir: {
  fake_sub_dir: {
    mount_a_real_dir: ...
    fake_file: ...
    another_fake_dir: ...
  }
});
```

I thought users were expecting the use cases like the above.

I guess your api support this?
```js
mockfs({ real_dir: mock.createDirectoryInfoFromPaths(...), fake_dir: ... });
```

Reply:
1. API Name

I followed convention which is generally adhered to for factories in TypeScript / JS code. You'll see this same format used in the TypeScript compiler, etc. The TypeScript type name of the config param that `mock()` takes is is `DirectoryInfo`, which is essentially an object that contains key: path, and value: content, DirectoryInfo, or a factory function which produces a `File` or `Directory` instance.

So this method creates such an object.

With that said, because this is a JS project, I agree that not all users may immediately recognize the name. Do you have a better name in mind?

2. Yes, you can easily merge automatic and manual

```
mock({
  ...mock.createDirectoryInfoFromPaths(...),
  'manual/path/': 'content'
});
```

I considered adding that to the readme. Should I?

Reply:
I might missunderstand your code. Yes, pls add that to readme.

If you simplify your api to read only one dir, then user can easily rename dir, and still easily load multiple dirs.

```js
mockfs({
  node_modules: mock.loadDir('../node_modules'),
  foo: mock.loadDir('bar')
  xx: {
    ...
  }
});
```

Reply:
I'll share my proposals followed by detail explaining them

1. Rename `mock.createDirectoryInfoFromPaths` to `mock.mapPaths()`
2. Add `mock.loadDir()`
3. Re-write the readme to clearly distinguish between automatic and manual mapping

## Automatic Mapping

The current factory generates a config object for full file paths. This is _automatic mapping_. The reason this is very useful is because it allows you to essentially create a shadow of specific parts of the actual file-system. Because it uses _lazy-loading_, we can now literally expose massive areas of your actual file-system without worrying that any modifications will take effect on the real hard drive.

This makes setting up the mock system as simple as `mock(mock.mapPaths([ path1, path2, path3 ]))`, and now all of those paths and their stat info is available.

## Manual mapping

For this, we have the existing functions, and I will also create a `mock.loadDir()`, which loads paths relatively and functions as you've shown.

What do you think of those proposals?

Reply:
I don't have strong opinion on the APIs, that sounds very good to me.
@tschaub may have better idea on what to do.

Reply:
Please dis-regard my previous comment. (I've deleted it)

## Current Proposal

1. Rename `mock.createDirectoryInfoFromPaths` to `mock.mapPaths()`
2. Add `mock.mapDir()` (same as your `mock.loadDir`)
3. Re-write the readme to clearly distinguish demonstrate automatic or manual mapping

## mock.mapPaths()

This is _automatic mapping_. It lets you create a full config map from one or more remote or relative paths.

Makes setting mock system as simple as `mock(mock.mapPaths('../'))` or `mock(mock.mapPaths([ '/srv/external-assets', '.' ])`, etc.

I'd recommend leaving this as it makes it simple and easy to quickly expose multiple 'mount points' on the filesystem. Due to lazy-loading, you can now expose very large areas.

## Manual mapping

This is for those who want to include some manual data. For this, we'll create `mock.mapDir()` that functions as you've shown.

```
mockfs({
  node_modules: mock.mapDir('../node_modules'),
  foo: mock.loadDir('bar', { recursive: false })
  xx: {
    ...
  }
});
```

What do you think on that?

Reply:
Great! I'll go ahead and make the changes for now. If @tschaub has any input, I'd be glad to hear it.